[WIP] Added React and Webpack files

biancadanforth opened the pull request:
Have the React Library and Webpack dev package installed, wired up the demo app into Doorhanger.jsm! Still haven't figured out how to get `require` to work, so I can include React components like `Cats` (from `cats.js`, obviously). :)

**Edit**: `require` is a webpack utility, so it gets removed by webpack in the build step. I need to think of an alternative way to include components! One way is to have a build step that does bundle afterall (though we don't have sourcemapping for debugging). One way is to have all the components for the app in a single page by design...

Also, having an HTML document inside a `<browser>` element be the entire panel does not display the dropdown menu properly (we get horizontal and vertical scroll bars, since with this method we have to explicitly specify the panel's width and height). Attempts to CSS hack the HTML `<select>` element have also come up short. This PR is still helpful in showing how React can be included in a bootstrapped addon.

biancadanforth committed:
[WIP] Added React and Webpack files, not yet integrated

biancadanforth committed:
Integrate React into Doorhanger.jsm

biancadanforth commented:
![screen shot 2017-10-13 at 4 49 28 pm](https://user-images.githubusercontent.com/17437436/31570291-390e88e6-b037-11e7-87ce-ec6700b2bdc7.png)

<img width="506" alt="screen shot 2017-10-13 at 4 57 05 pm" src="https://user-images.githubusercontent.com/17437436/31570338-8262f34c-b037-11e7-8c54-65d2260ecb36.png">

gregglind commented on addon/content/cats/app.js:
```diff
@@ -1,15 +1,45 @@
+  const r = React.createElement;
+
+  // TODO bdanforth: 'require' is not importing the cats.js component correctly
+  // const Cats = require("resource://focused-cfr-shield-study-content/cats/cats.js");
```
So things:  
1.  Require isn't going to work in firefox code *AT ALL*. 
2.  during the 'webpack' phase, 'resource' doesn't exist.
3.  During the firefox phase, require doesn't exist.

Options:
1.  You bundle all the resources in during packing using their relative (node like) require semantics.

biancadanforth committed:
Add bundling and JSX to cats subfolder

Now there is a Cats component rendered inside of an App component using JSX.
Added a second webpack config to bundle these React components into a single file, currently called 'UI.js' inside the 'cats' folder.

biancadanforth committed:
Improve npm run firefox

biancadanforth committed:
Add select element dropdown for demo purposes
